### PR TITLE
Ft11 annotation configs

### PR DESCRIPTION
**Pull request by benjamin-confino**

### CODE REVIEW

**Comment by benjamin-confino** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.microprofile.faulttolerance.cdi_fat,com.ibm.ws.microprofile.faulttolerance_fat_tck,com.ibm.ws.microprofile.faulttolerance.cdi_fat_metrics

File: dev/com.ibm.ws.microprofile.faulttolerance.cdi/src/com/ibm/ws/microprofile/faulttolerance/cdi/FaultToleranceCDIExtension.java [hunk 1]
```diff
@@ -101,6 +101,9 @@ public void beforeBeanDiscovery(@Observes BeforeBeanDiscovery beforeBeanDiscover
                     BulkheadConfig bulkhead = new BulkheadConfig(clazz, (Bulkhead) annotation);
                     bulkhead.validate();
                 }
+            } else { 
+                if (tc.isWarningEnabled())
+                    Tr.warning(tc, "Annotation {0} on {1} was disabled and will be ignored", annotation.annotationType().getSimpleName(), clazz.getCanonicalName());
```
Comment (tevans78) [inline dev/com.ibm.ws.microprofile.faulttolerance.cdi/src/com/ibm/ws/microprofile/faulttolerance/cdi/FaultToleranceCDIExtension.java]: Warnings should be NLS'd

File: dev/com.ibm.ws.microprofile.faulttolerance.cdi/src/com/ibm/ws/microprofile/faulttolerance/cdi/FaultToleranceInterceptor.java [hunk 2]
```diff
@@ -111,8 +111,9 @@ private AggregatedFTPolicy processPolicies(InvocationContext context, BeanManage
         Class<?> targetClass = context.getTarget().getClass();
         Annotation[] annotations = targetClass.getAnnotations();
         for (Annotation annotation : annotations) {
-            // Don't process any annotations which aren't enabled
-            if (!FTGlobalConfig.getActiveAnnotations(targetClass).contains(annotation.annotationType())) {
+
+            //Check that the annotation has not been disabled for this specific class. 
+            if (! FTGlobalConfig.isAnnotationEnabled(annotation, targetClass)) {
```
Comment (tevans78) [inline dev/com.ibm.ws.microprofile.faulttolerance.cdi/src/com/ibm/ws/microprofile/faulttolerance/cdi/FaultToleranceInterceptor.java]: I don't really like this continue ... why not remove the negation and just put the whole thing inside the if block?

File: dev/com.ibm.ws.microprofile.faulttolerance.cdi/src/com/ibm/ws/microprofile/faulttolerance/cdi/config/FTGlobalConfig.java [hunk 3]
```diff
@@ -43,6 +46,73 @@
...
+    public static boolean isAnnotationEnabled(Annotation ann, Class<?> clazz) {
+        return isAnnotationEnabled(ann, clazz, null);
+    }
+
+    /**
+     * Checks if an annotation is enabled or disabled via configuration options. Annotations can be disabled with the following syntax:
+     *
+     * <ul>
+     * <li>Disable on the class-level: com.acme.test.MyClient/serviceA/CircuitBreaker/enabled=false</li>
+     * <li>Disable at the method-level: com.acme.test.MyClient/serviceA/methodA/CircuitBreaker/enabled=false</li>
```
Comment (tevans78) [inline dev/com.ibm.ws.microprofile.faulttolerance.cdi/src/com/ibm/ws/microprofile/faulttolerance/cdi/config/FTGlobalConfig.java]: swap class and method here to reflect the priority

File: dev/com.ibm.ws.microprofile.faulttolerance.cdi/src/com/ibm/ws/microprofile/faulttolerance/cdi/config/FTGlobalConfig.java [hunk 4]
```diff
@@ -43,6 +46,73 @@
...
+        // Find the real class since we probably have a Weld proxy
+        clazz = FTUtils.getRealClass(clazz);
+        ClassLoader cl = FTUtils.getClassLoader(clazz);
+        Config mpConfig = ConfigProvider.getConfig(cl);
+
+        if (method != null) {
+            String methodKey = clazz.getCanonicalName() + "/" + method.getName() + "/" + ann.annotationType().getSimpleName() + "/enabled";
+            Optional<Boolean> methodEnabled = mpConfig.getOptionalValue(methodKey, Boolean.class);
+            if (methodEnabled.isPresent()) {
+                return methodEnabled.get(); //Method scoped properties take precedence. We can return, otherwise move on to class scope.
```
Comment (tevans78) [inline dev/com.ibm.ws.microprofile.faulttolerance.cdi/src/com/ibm/ws/microprofile/faulttolerance/cdi/config/FTGlobalConfig.java]: You know I'm not a fan of multiple return statements like this!

**Pull request merged**